std/int128.d

```diff
@@ -147,13 +147,58 @@ public struct Int128
 
     /** Support casting to a bool
      * Params: T = bool
-     * Returns: boolean result
+     * Returns: true if value is not zero
      */
     bool opCast(T : bool)() const
     {
         return tst(this.data);
     }
 
+    /** Support casting to an integral type
+     * Params: T = integral type
+     * Returns: low bits of value reinterpreted as T
+     */
+    T opCast(T : long)() const
+    if (is(byte : T))
+    {
+        return cast(T) this.data.lo;
+    }
+
+    ///
+    @safe unittest
+    {
+        const Int128 a = Int128(0xffff_ffff_ffff_ffffL, 0x0123_4567_89ab_cdefL);
+        assert(cast(long) a == 0x0123_4567_89ab_cdefL);
+        assert(cast(int)  a ==           0x89ab_cdef);
+        assert(cast(byte) a == cast(byte) 0xef);
+    }
+
+    /** Support casting to floating point type
+     * Params: T = floating point type
+     * Returns: value cast to floating point with environment-dependent
+     * rounding if the value is not exactly representable
+     */
+    T opCast(T : real)() const
+    if (is(real : T))
```

Comment:
This constraint looks redundant with the `T : real` specialization

Reply:
Changed.